first commit

jonathansmit7 opened the pull request:
@bastbijl code check

Commit:
first commit

bastbijl commented on pom.xml:
```diff
@@ -0,0 +1,50 @@
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>weaver-xmi-importer</groupId>
+    <artifactId>weaver-xmi-importer</artifactId>
```
zou je dit gelijk willen trekken met de naam van de repo nu, dus weaver-import-xmi

Commit:
pom: group id and artifact id changed to repo name. package name changed.

gvdent commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      xmi = xmi.replaceAll("UML:", "UML.");
```
From here on some XMI processing appears to happen. This functionality should be extracted to its own method, as the main is getting quite long and complex, extracting functionality should make it more clear what is going on. (3 nested for loops make it pretty hard to see at a glance what is going on, this is a code smell).

gvdent commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+      try {
+        xmi = IOUtils.toString(xmiImporter.read());
+      } catch (IOException e) {
+        e.printStackTrace();
```
Program should exit on this exception I guess, continuing here seems useless. See comment later in file.

gvdent commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+      }
+
+    }catch(Exception e) {
+      System.out.println("cannot read!");
```
Catch more specific exceptions, do not eat exceptions in general. Maybe wrap the exception in a RuntimeException so the application will exit, as proceeding when an exception is thrown here seems pretty useless.

malamili commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+ */
+public class XmiImporter {
+
+  private Weaver WEAVER;
```
Add final if constant, otherwise do not use CAPITAL.

malamili commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+  public XmiImporter(Weaver weaver, String file, boolean useUnixPath){
+    WEAVER = weaver;
+    FILE = file;
+    USE_UNIX_FILEPATH = useUnixPath;
```
Why is this boolean necessary?

malamili commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+      XML doc = new XMLDocument(xmi);
+      List<XML> nodeClasses = doc.nodes(xpath);
+
+      HashMap<String, String> aIndividualMap = new HashMap<>();
```
Just name this map 'individuals' or 'individualMap', the 'a' is not helping understanding what this is

malamili commented on src/main/java/com/weaverplatform/import/xmi/XmiImporter.java:
```diff
@@ -0,0 +1,270 @@
+      }
+      xmi = xmi.replaceAll("UML:", "UML.");
+
+      String xpath = "//XMI.content/UML.Model/UML.Namespace.ownedElement/UML.Package/UML.Namespace.ownedElement/UML.Package/UML.Namespace.ownedElement/UML.Class";
```
Give this xpath (and consequent xpaths) a more meaninfull name, what does it point to?

Commit:
refactored classes.

Commit:
refactored gone

Commit:
refactored classes update

Commit:
refactored main and update classes

Commit:
comments added in Main-function

jonathansmit7 commented:
now updated!

Commit:
more tests added

Commit:
added: getFormattedFileContents, getFormattedXML methods.
This way responsibility spread over multiple methods. Make Test-class ready for new methods.
Comments added to main class.

Commit:
refactor for main. added: run method. als class description.

Commit:
comments for method run

Commit:
refactored, clean up tests

Commit:
refactored method

Commit:
refactor method

Commit:
refactor methods

Commit:
changing some comments

gvdent commented:
Much cleaner, thanks for the improvements! Merge on!

Commit:
refactoring methods

Commit:
comment moved from method

Commit:
run method update

Commit:
update NotNull method

Commit:
update dataset link

Commit:
update voor entity link definition, entity type definition: now using the right sdk-classes.

Commit:
update with usage relationkey-class

Commit:
weaver connect with weaver socket class, added a comment for third static main argument

Commit:
latest update

Commit:
- Added support for InputStream (for NiFi)
- Refactored and cleaned some code
- Made pom.xml ready for Central release
